lib/core/src/sync/mod.rs

```diff
@@ -244,40 +256,97 @@ impl SyncService {
                 .map(|state| state.is_local)
                 .unwrap_or(false),
         };
-        let is_update = maybe_sync_state.is_some();
         let last_commit_time = maybe_outgoing_changes.map(|details| details.commit_time);
-        self.commit_record(
-            &decrypted_record,
+
+        Ok(DecryptionInfo {
             new_sync_state,
-            is_update,
+            record: decrypted_record,
             last_commit_time,
-        )?;
+        })
+    }
 
-        log::info!(
-            "Successfully pulled record record_id {}",
-            &decrypted_record.id
-        );
+    async fn handle_recovery(
+        &self,
+        swap_decryption_info: Vec<DecryptionInfo>,
+    ) -> Result<Vec<(DecryptionInfo, Swap)>> {
+        let mut succeded = vec![];
+        let mut swaps = vec![];
+
+        // Step 1: Convert each record into a swap, if possible
+        for decryption_info in swap_decryption_info {
+            let record = &decryption_info.record;
+            match TryInto::<Swap>::try_into(record.data.clone()) {
+                Ok(swap) => {
+                    succeded.push(decryption_info);
+                    swaps.push(swap);
```

Comment:
It gets a bit tricky later on when we have to do this:
https://github.com/breez/breez-sdk-liquid/blob/7638593821cc77a2c46683e3f77a87c0ce07149c/lib/core/src/sync/mod.rs#L291-L293
Since we'd have to take a mutable slice of all the second elements in the tuple. Looks a little more complicated than splitting the two, but can be done

Reply:
Just questioning, keep it your way if its simpler